controllers/clusterresources/awssecuritygroupfirewallrule_controller.go

```diff
@@ -74,16 +74,17 @@ func (r *AWSSecurityGroupFirewallRuleReconciler) Reconcile(ctx context.Context,
 		return models.ReconcileRequeue, err
 	}
 
-	switch firewallRule.Annotations[models.ResourceStateAnnotation] {
-	case models.CreatingEvent:
-		reconcileResult := r.handleCreateFirewallRule(ctx, firewallRule, &l)
-		return reconcileResult, nil
-	case models.DeletingEvent:
-		reconcileResult := r.handleDeleteFirewallRule(ctx, firewallRule, &l)
-		return reconcileResult, nil
-	case models.GenericEvent:
+	if firewallRule.Status.ClusterEvent == models.CreatingEvent {
+		return r.handleCreateFirewallRule(ctx, firewallRule, &l), nil
+	}
+
+	if firewallRule.Status.ClusterEvent == models.DeletingEvent {
+		return r.handleDeleteFirewallRule(ctx, firewallRule, &l), nil
+	}
+
+	if firewallRule.Annotations[models.ResourceStateAnnotation] == models.GenericEvent {
 		l.Info("AWS security group firewall rule event isn't handled",
-			"cluster ID", firewallRule.Spec.ClusterID,
+			"cluster ID", firewallRule.Status.ClusterID,
 			"type", firewallRule.Spec.Type,
 			"request", req,
 			"event", firewallRule.Annotations[models.ResourceStateAnnotation])
```

Comment:
Do we really need to handle the `Generic Event` if we rely on resource status?

Reply:
agree fixed